### PR TITLE
[IMP] General: Update manage inbound email doc

### PR DESCRIPTION
**Pull request by jero-odoo**
Docs task: https://www.odoo.com/mail/view?model=project.task&res_id=4171039&access_token=317cf3e9-8e53-4162-92bc-8770f77370ae

Updating V17 with sections on "Allow alias domain system parameter" and "Local-part based incoming detection" as per docs task.

### CODE REVIEW

**Comment by jero-odoo** [comment]
Hi @ksc-odoo  this is ready for peer review. (I only updated lines 321-360 for this pr)
Thanks!

File: content/applications/general/email_communication/email_servers_inbound.rst [hunk 1]
```diff
@@ -317,3 +317,44 @@ To change the default behavior, enable :ref:`developer-mode`, then go to :menuse
...
+for which the alias can create a ticket, lead, opportunity, etc., eliminates false positives where
+email addresses with only the prefix alias, not the domain are present.
+
+In some instances, matches have been made in the Odoo database when an email is received with the
+same alias prefix and a different domain on the incoming email address. This is true in the sender,
+recipient, and :abbr:`CC (Carbon Copy)` email addresses of an incoming email.
+
+.. example::
+   When Odoo receives emails that have the name `commercial` prefix alias in the sender, recipient,
+   or :abbr:`CC (Carbon Copy)` email addresses (e.g. commercial@gmail.com, commercial@odoo.net),
```
Comment (ksc-odoo) [inline content/applications/general/email_communication/email_servers_inbound.rst:339]:
```suggestion
   or :abbr:`CC (Carbon Copy)` email addresses (e.g. commercial@gmail.com),
```

File: content/applications/general/email_communication/email_servers_inbound.rst [hunk 2]
```diff
@@ -317,3 +317,44 @@ To change the default behavior, enable :ref:`developer-mode`, then go to :menuse
...
+:guilabel:`Save`, and the system parameter takes immediate effect.
+
+.. image:: email_servers_inbound/allowed-domain.png
+   :alt: mail.catchall.domain.allowed system parameter set.
+
+Local-part based incoming detection
+===================================
+
+When creating a new alias, there is an option to enable :guilabel:`Local-part based incoming
+detection`. If enabled, Odoo will only require the Local-part to match for routing an incoming
```
Comment (ksc-odoo) [inline content/applications/general/email_communication/email_servers_inbound.rst:358]:
```suggestion
detection`. If enabled, Odoo only requires the Local-part to match for routing an incoming
```

**Review by ksc-odoo — APPROVED** [review]
@jero-odoo -- just finished my Final Review. Good work. Once you implement the few suggestions I've left, feel free to move this to Tech Review. Thanks!

File: content/applications/general/email_communication/email_servers_inbound.rst [hunk 3]
```diff
@@ -317,3 +317,44 @@ To change the default behavior, enable :ref:`developer-mode`, then go to :menuse
...
+   the database falsely treats the email as the full `commercial` alias, with a different domain,
+   and therefore, creates a ticket/lead/opportunity/etc.
+
+To add the `mail.catchall.domain.allowed` system parameter, first, activate the :ref:`developer mode
+<developer-mode>`. Then, go to :menuselection:`Settings app --> Technical --> Parameters section -->
+System Parameters`. Click :guilabel:`New`. Then, type in `mail.catchall.domain.allowed` for the
+:guilabel:`Key` field.
+
+Next, for the :guilabel:`Value` field, add the domains separated by commas. Manually
+:guilabel:`Save`, and the system parameter takes immediate effect.
```
[the remark this answers is not in the document]
Comment (StraubCreative) [reply content/applications/general/email_communication/email_servers_inbound.rst]: @jero-odoo i think you had it correct the first time (please check), just would need to add the ☁️ icon with it

[the remark this answers is not in the document]
Comment (samueljlieber) [reply content/applications/general/email_communication/email_servers_inbound.rst]: Agreed, how @jero-odoo had it the first time is almost correct– just need to add the icon and the "Save" descriptor in parenthesis. 

Even though the tooltip says "Save manually" on hover, "Save" is what was agreed on :)
```rst
Next, for the :guilabel:`Value` field, add the domains separated by commas. Manually
:icon:`fa-cloud-upload` :guilabel:`(Save)`, and the system parameter takes immediate effect.
```

**Comment by jero-odoo** [comment]
@samueljlieber this is ready for tech review. Thanks!

File: content/applications/general/email_communication/email_servers_inbound.rst [hunk 4]
```diff
@@ -317,3 +317,44 @@ To change the default behavior, enable :ref:`developer-mode`, then go to :menuse
...
+:icon:`fa-cloud-upload` :guilabel:`(Save)`, and the system parameter takes immediate effect.
+
+.. image:: email_servers_inbound/allowed-domain.png
+   :alt: mail.catchall.domain.allowed system parameter set.
+
+Local-part based incoming detection
+===================================
+
+When creating a new alias, there is an option to enable :guilabel:`Local-part based incoming
+detection`. If enabled, Odoo only requires the Local-part to match for routing an incoming email. If
```
Comment (samueljlieber) [inline content/applications/general/email_communication/email_servers_inbound.rst:358]: IMO this shouldnt be capitalized
```suggestion
detection`. If enabled, Odoo only requires the local-part to match for routing an incoming email. If
```

**Review by samueljlieber — APPROVED** [review]
Hi @jero-odoo, nice job on this PR. Your improvements look good, I just have a couple of technical changes :) 

Some food for thought- I would also advocate to update this entire doc for clarity and flow, as there are some instances where instructions are relying on images for context, and this doc seems to be missing a general overview on what an email alias actually is.

Thank you for your work!
..
@robodoo delegate=jero-odoo

**Comment by jero-odoo** [comment]
Thank you Sam! Will look into adding this to the roadmap.

@robodoo r+

**Pull request closed without merging**